ui/raidboss/popup-text.js

```diff
@@ -745,10 +748,26 @@ class PopupText {
       triggerHelper.trigger.run(this.data, triggerHelper.matches);
   }
 
-  _addText(container, e) {
-    container.appendChild(e);
-    if (container.children.length > this.kMaxRowsOfText)
-      container.removeChild(container.children[0]);
+  _createTextFor(text, textType, lowerTextKey, duration) {
+    // info-text
+    let textElementClass = textType + '-text';
+    if (textType !== 'info')
+      text = triggerUpperCase(text);
+    let holder = this[lowerTextKey].getElementsByClassName('holder')[0];
+    let div = this._makeTextElement(text, textElementClass);
+
+    holder.appendChild(div);
+    if (holder.children.length > this.kMaxRowsOfText)
+      holder.removeChild(holder.children[0]);
+
+    window.setTimeout(() => {
+      for (let i = 0; i < holder.children.length; ++i) {
+        if (holder.children[i] == e) {
```

Comment:
`e` isn't defined here anymore.  I think this loop is trying to say `holder.removeChild(div)`?

Reply:
Actually, that's a good point. Couldn't that entire block be rewritten as this?
```javascript
    window.setTimeout(() => {
      holder.removeChild(div);
    }, duration * 1000);
```

Reply:
Hmm, no, that would throw an exception if the timeline/etc reset before the timeout.

Reply:
Ah, that's fair.